homeassistant/components/flashforge/__init__.py

```diff
@@ -0,0 +1,44 @@
+"""The FlashForge 3D Printer integration."""
+from __future__ import annotations
+
+from ffpp.Printer import Printer
+
+from homeassistant.components.sensor import DOMAIN as SENSOR_DOMAIN
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_IP_ADDRESS, CONF_PORT
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN, SCAN_INTERVAL
+from .data_update_coordinator import FlashForgeDataUpdateCoordinator
+
+PLATFORMS = [SENSOR_DOMAIN]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up FlashForge 3D Printer from a config entry."""
+    printer = Printer(entry.data[CONF_IP_ADDRESS], port=entry.data[CONF_PORT])
+
+    hass.data.setdefault(DOMAIN, {})
+
+    coordinator = FlashForgeDataUpdateCoordinator(hass, printer, entry, SCAN_INTERVAL)
+
+    await coordinator.async_config_entry_first_refresh()
+
+    # Save the printer, and coordinator object to be able to access it later on.
+    hass.data[DOMAIN][entry.entry_id] = {
+        "printer": printer,
+        "coordinator": coordinator,
+    }
```

Comment:
Not required and perhaps not directly for this PR, but you could implement a NamedTuple to type this data. This will also make it easier to access this data in your other classes.

https://github.com/home-assistant/core/blob/efddace53ac3421f44b6d302cbd8ab5f7e982092/homeassistant/components/tuya/__init__.py#L44-L49

Reply:
I have been thinking about the data dictionary i use today and this is nicer. Since I'm planning to implement more platforms (camera/perhaps some service to pause/abort a 3d print) I'll probably refactor this in a later on.